File: awoo-bot/awoo-bot.js
const { Client, Intents, GatewayIntentBits, ActivityType, Guild } = require('discord.js');
const fs = require('fs');
const { REST } = require('@discordjs/rest');
const { Routes } = require('discord-api-types/v9');

//@xkawaimugi Load configuration from config.json
const config = require('./config/config.json');
const { clientId, guildIds, token, userId } = config;

const client = new Client({
  intents: [GatewayIntentBits.Guilds, GatewayIntentBits.GuildMessages],
});

async function setBotStatus(client, guildIds) {
  try {
    for (const guildId of guildIds) {
      const guild = await client.guilds.fetch(guildId);
      const user = guild.members.resolve(client.user.id);

      await client.user.setPresence({
        activities: [
          {
            name: 'mugi.me | /awoo',
            type: ActivityType.Playing
          }
        ],
        status: 'online'
      });

      console.log(`Bot status set successfully for guild: ${guildId}`);
    }
  } catch (error) {
    console.error('Failed to set bot status:', error);
  }
}

client.once('ready', () => {
  console.log(`Logged in as ${client.user.tag}!`);

  // Set the initial bot status
  setBotStatus(client, guildIds);
});

const rest = new REST({ version: '9' }).setToken(token);

(async () => {
  try {
    console.log('Started refreshing application (/) commands.');

    const commands = [];
    const commandFiles = fs.readdirSync('./commands').filter((file) => file.endsWith('.js'));

    for (const file of commandFiles) {
      const command = require(`./commands/${file}`);
      commands.push(command.data.toJSON());
    }

    for (const guildId of guildIds) {
      await rest.put(Routes.applicationGuildCommands(clientId, guildId), { body: commands });
      console.log(`Registered slash commands for guild: ${guildId}`);
    }

    console.log('Successfully registered application (/) commands.');
  } catch (error) {
    console.error('Failed to register application (/) commands:', error);
  }
})();

client.commands = new Map();
const commandFiles = fs.readdirSync('./commands').filter((file) => file.endsWith('.js'));
for (const file of commandFiles) {
  //commands @xkawaimugi
  const command = require(`./commands/${file}`);
  client.commands.set(command.data.name, command);
}

client.on('interactionCreate', async (interaction) => {
  if (!interaction.isCommand()) return;

  const { commandName } = interaction;
  const command = client.commands.get(commandName);

  if (!command) return;

  try {
    await command.execute(interaction);
  } catch (error) {
    console.error(`Error executing command ${commandName}:`, error);
    await interaction.reply({ content: 'An error occurred while executing this command.', ephemeral: true });
  }
});

client.login(token);

File: awoo-bot.py/awoo_bot.py
import discord
from discord.ext import commands
from discord import app_commands
import interactions
import json
import os

# Load configuration from config.json
with open('./config/config.json') as config_file:
    config = json.load(config_file)
    guildIds = config['guildIds']
    token = config['token']

# Set up the client with intents
intents = discord.Intents.all()
intents.guilds = True
intents.messages = True

client = commands.Bot(command_prefix='/', intents=intents)


async def set_bot_status():
    for guild_id in guildIds:
        guild = await client.fetch_guild(guild_id)
        if guild:
            await client.change_presence(activity=discord.Game(name='mugi.pages.dev | /awoo'))
            print(f'Bot status set successfully for guild: {guild_id}')


@client.event
async def on_interaction(interaction: discord.Interaction):
    if interaction.type == discord.InteractionType.application_command:
        command = client.get_application_command(interaction.command.name)
        if command:
            try:
                await command.callback(interaction)
            except Exception as e:
                print(f'Error executing command {interaction.command.name}:', e)
                await interaction.response.send_message('An error occurred while executing this command.',
                                                        ephemeral=True)


async def load_commands():
    print('Started refreshing application (/) commands.')

    # Load commands from files
    for filename in os.listdir('./commands'):
        if filename.endswith('.py') and not filename.startswith('__'):
            extension = filename[:-3]
            try:
                await client.load_extension(f'commands.{extension}')
                print(f'Successfully loaded command: {extension}')
            except Exception as e:
                print(f'Failed to load command {extension}: {str(e)}')

    print('Successfully registered application (/) commands.')


@client.event
async def on_ready():
    await client.user.edit(username='awoo')
    print(f'Logged in as {client.user}!')
    await load_commands()
    await set_bot_status()


client.run(token)
